threshsign/src/ThresholdSignaturesTypes.cpp

```diff
@@ -15,11 +15,18 @@
 #include "threshsign/ThresholdSignaturesTypes.h"
 #include "threshsign/IThresholdSigner.h"
 #include "threshsign/IThresholdVerifier.h"
-#include "threshsign/bls/relic/BlsThresholdFactory.h"
-#include "threshsign/bls/relic/PublicParametersFactory.h"
+#include "threshsign/IThresholdFactory.h"
 #include "yaml_utils.hpp"
 #include "Logger.hpp"
+
+#ifdef USE_RELIC
+#include "threshsign/bls/relic/BlsThresholdFactory.h"
+#include "threshsign/bls/relic/PublicParametersFactory.h"
+#endif
+
+#ifdef USE_EDDSA_OPENSSL
```

Comment:
`#elseif` ?

Reply:
I know, it's on the cmake level, here is a source code level

Reply:
I added a header which hides this #ifdef
The general problem is that there needs either be a file which knows both implementations (and future implementations)
or an inheritance hierarchy for cryptosystem.
I chose to think about cryptosystem as a factory, which might be the wrong thing to do, but it was quickest.